qemu/tests/qemu_guest_agent.py

```diff
@@ -827,26 +865,40 @@ def setup_gagent_in_host(self, params, vm):
         if not os.path.exists(gagent_host_path):
             raise error.TestFail("qemu-ga install program is not exist, maybe "
                                  "the program is not successfully downloaded ")
-        session = self._get_session(params, vm)
         gagent_guest_dir = params["gagent_guest_dir"]
 #        gagent_remove_service_cmd = params["gagent_remove_service_cmd"]
         s, o = session.cmd_status_output("mkdir %s" % gagent_guest_dir)
         if bool(s) and str(s) != "1":
             raise error.TestError("Could not create qemu-ga directory in "
                                   "VM '%s', detail: '%s'" % (vm.name, o))
-        error.context("Copy qemu guest agent program to guest", logging.info)
+        error_context.context("Copy qemu guest agent program to guest", logging.info)
         vm.copy_files_to(gagent_host_path, gagent_guest_dir)
 
+    @error_context.context_aware
     def setup(self, test, params, env):
```

Comment:
Not need to keep this since it has nothing difference with the one in base class.

Reply:
The difference is for windows guest , need setup in host part.
self.setup_gagent_in_host(session, params, self.vm)

Reply:
@luckyh  
I update this patch according to your advice, could you help review it?
Thanks.